packages/typescript/internal/ts_project.bzl

```diff
@@ -274,21 +325,58 @@ def _replace_ext(f, ext_map):
         return new_ext
     return None
 
-def _out_paths(srcs, outdir, rootdir, allow_js, ext_map):
-    rootdir_replace_pattern = rootdir + "/" if rootdir else ""
+def _out_paths(srcs, out_dir, root_dir, allow_js, ext_map):
+    rootdir_replace_pattern = root_dir + "/" if root_dir else ""
     outs = []
     for f in srcs:
         if _is_ts_src(f, allow_js):
-            out = _join(outdir, f[:f.rindex(".")].replace(rootdir_replace_pattern, "") + _replace_ext(f, ext_map))
+            out = _join(out_dir, f[:f.rindex(".")].replace(rootdir_replace_pattern, "") + _replace_ext(f, ext_map))
 
             # Don't declare outputs that collide with inputs
             # for example, a.js -> a.js
             if out != f:
                 outs.append(out)
     return outs
 
+def _calculate_js_outs(srcs, out_dir, root_dir, allow_js, preserve_jsx, emit_declaration_only):
```

Comment:
nit: I think it's a bit more descriptive to name these functions "predict_*_outs" rather than calculate...

Reply:
This function is called twice, once in the macro to determine if outputs can be predeclared, and again inside the rule impl to determine the full list of output files. In the first case, I would agree that `predict` accurately describes it, but in the rule implementation it yields a full list of the outputs that will be generated by running node, so I would argue it's not making a prediction anymore at that point. I don't have a strong opinion on the naming here though, so if you prefer it to use the verb predict I'm okay with that.